src/CheckoutSdk/ApiClient.cs

```diff
@@ -122,11 +123,11 @@ public async Task<TResult> PostAsync<TResult>(string path, IApiCredentials crede
                 idempotencyKey: idempotencyKey
                 ))
             {
-                return await DeserializeJsonAsync<TResult>(httpResponse);
+                return await httpResponse.ConvertToChekoutHttpResponseMessage<TResult>();
             }
         }
 
-        public async Task<dynamic> PostAsync(string path, IApiCredentials credentials, Dictionary<HttpStatusCode, Type> resultTypeMappings, CancellationToken cancellationToken, object request = null, string idempotencyKey = null)
+        public async Task<CheckoutHttpResponseMessage<dynamic>> PostAsync(string path, IApiCredentials credentials, Dictionary<HttpStatusCode, Type> resultTypeMappings, CancellationToken cancellationToken, object request = null, string idempotencyKey = null)
```

Comment:
I think this method might be the source of your test failures for `Microsoft.CSharp.RuntimeBinder.RuntimeBinderException : The best overloaded method match for 'Checkout.CheckoutHttpResponseMessage<Checkout.Payments.CaptureResponse>.CheckoutHttpResponseMessage(System.Net.HttpStatusCode, System.Net.Http.Headers.HttpResponseHeaders, Checkout.Payments.CaptureResponse)' has some invalid arguments`

there seems to be a bit too much dynamic going on here - can you point me to the method where this particular overload is used? I'm having a bit of hard time navigating in this PR, and currently don't have a VS Studio on my laptop I could use

Reply:
It's being used in 2 locations
- https://github.com/checkout/checkout-sdk-net/pull/102/files#diff-4268ee75d24bfad7856e52ec5739825f2b4d8a58919216607dfca67b4babecb0R79
- https://github.com/checkout/checkout-sdk-net/pull/102/files#diff-5c372cea63939a75da112d4449f741b0821dd313796ad4fa8958b558c509ec31R47

Reply:
Thanks - I think in general this looks overly abstracted and can most likely be simplified, removing the need for `dynamic` altogether.

E.g. https://github.com/checkout/checkout-sdk-net/pull/102/files#diff-5c372cea63939a75da112d4449f741b0821dd313796ad4fa8958b558c509ec31R47

has only one single entry for the SourceResponseMappings, so I wonder why it needs a mapping at all? 

I'll give this another look a bit later, but if you can try and see if you can remove these generalisations I think it'd be a good start